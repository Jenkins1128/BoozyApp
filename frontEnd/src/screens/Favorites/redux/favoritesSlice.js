import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';
const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get(`${baseUrl}/get`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
