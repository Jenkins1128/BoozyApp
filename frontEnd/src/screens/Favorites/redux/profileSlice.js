import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/get');
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
