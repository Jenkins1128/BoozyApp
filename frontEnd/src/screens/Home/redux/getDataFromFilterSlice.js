import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const getDataFromFilterAsync = createAsyncThunk('getDataFromFilterAsync/status', async (data, { rejectWithValue }) => {
	const { location, priceType, cuisine } = data;
	try {
		const response = await axios.get(`${baseUrl}/yelp/${location}/${priceType}/${cuisine}`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getDataFromFilterSlice = createSlice({
	name: 'getDataFromFilterAsync',
	initialState,
	reducers: {
		resetDataFromFilterRequestStatus: (state) => {
			state.currentState.dataFromFilterRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getDataFromFilterAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				dataFromFilterRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getDataFromFilterAsync.rejected, (state) => {
			state.currentState.dataFromFilterRequestStatus = 'rejected';
		});
	}
});

export const { resetDataFromFilterRequestStatus } = getDataFromFilterSlice.actions;
export const selectDataFromFilterState = (state) => state.dataFromFilter.currentState;
export default getDataFromFilterSlice.reducer;
