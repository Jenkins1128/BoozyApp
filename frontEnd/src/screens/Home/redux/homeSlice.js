import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {
		location: '',
		cuisine: null,
		priceType: 6,
		$color: 'white',
		$$color: 'white',
		$$$color: 'white',
		$$$$color: 'white',
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: [],
		initialPosition: {
			latitude: 0,
			longitude: 0
		},
		markerPosition: {
			latitude: 0,
			longitude: 0
		}
	}
};

export const getCurrentLocationDataAsync = createAsyncThunk('getCurrentLocationDataAsync/status', async (data, { rejectWithValue }) => {
	const { latitude, longitude } = data;
	try {
		const response = await axios.get(`${baseUrl}/yelp/${latitude}/${longitude}`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const homeSlice = createSlice({
	name: 'getCurrentLocationDataAsync',
	initialState,
	reducers: {
		updateLocation: (state, { payload }) => {
			state.currentState.location = payload.location;
		},
		updateCuisine: (state, { payload }) => {
			state.currentState.cuisine = payload.cuisine;
		},
		updatePriceType: (state, { payload }) => {
			state.currentState.priceType = payload.priceType;
		},
		update$color: (state, { payload }) => {
			state.currentState.$color = payload.$color;
		},
		update$$color: (state, { payload }) => {
			state.currentState.$$color = payload.$$color;
		},
		update$$$color: (state, { payload }) => {
			state.currentState.$$$color = payload.$$$color;
		},
		update$$$$color: (state, { payload }) => {
			state.currentState.$$$$color = payload.$$$$color;
		},
		updateAllCategories: (state, { payload }) => {
			state.currentState.allCategories = payload.allCategories;
		},
		updateAlreadyFavorited: (state, { payload }) => {
			state.currentState.alreadyFavorited = payload.alreadyFavorited;
		},
		updateAlreadyRated: (state, { payload }) => {
			state.currentState.alreadyRated = payload.alreadyRated;
		},
		updateRestaurantsArray: (state, { payload }) => {
			state.currentState.restaurantsArray = payload.restaurantsArray;
		},
		updateInitialPosition: (state, { payload }) => {
			state.currentState.initialPosition = payload.initialPosition;
		},
		updateMarkerPosition: (state, { payload }) => {
			state.currentState.markerPosition = payload.markerPosition;
		},
		reset: (state) => {
			Object.assign(state.currentState, {
				location: '',
				cuisine: null,
				priceType: 6,
				$color: 'white',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice1: (state) => {
			Object.assign(state.currentState, {
				priceType: 1,
				$color: 'yellow',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice2: (state) => {
			Object.assign(state.currentState, {
				priceType: 2,
				$color: 'white',
				$$color: 'yellow',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice3: (state) => {
			Object.assign(state.currentState, {
				priceType: 3,
				$color: 'white',
				$$color: 'white',
				$$$color: 'yellow',
				$$$$color: 'white'
			});
		},
		setPrice4: (state) => {
			Object.assign(state.currentState, {
				priceType: 4,
				$color: 'white',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'yellow'
			});
		},
		resetHomeRequestStatus: (state) => {
			state.currentState.locationDataRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(getCurrentLocationDataAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					locationDataRequestStatus: 'fulfilled',
					restaurantsArray: payload
				});
			})
			.addCase(getCurrentLocationDataAsync.rejected, (state) => {
				state.currentState.locationDataRequestStatus = 'rejected';
			});
	}
});

export const {
	updateLocation,
	updateCuisine,
	updatePriceType,
	update$color,
	update$$color,
	update$$$color,
	update$$$$color,
	updateAllCategories,
	updateAlreadyFavorited,
	updateAlreadyRated,
	updateRestaurantsArray,
	updateInitialPosition,
	updateMarkerPosition,
	reset,
	setPrice1,
	setPrice2,
	setPrice3,
	setPrice4,
	resetHomeRequestStatus
} = homeSlice.actions;

export const selectHomeState = (state) => state.home.currentState;
export default homeSlice.reducer;
