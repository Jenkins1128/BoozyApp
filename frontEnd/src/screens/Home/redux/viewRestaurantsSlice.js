import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const viewRestaurantsAsync = createAsyncThunk('viewRestaurantsAsync/status', async (data, { rejectWithValue }) => {
	const { index, state } = data;
	state.index = index;
	let restaurantId;
	if (state.restaurantsArray[index]['id']) {
		restaurantId = state.restaurantsArray[index]['id'];
	} else if (state.restaurantsArray[index]['restaurantId']) {
		restaurantId = state.restaurantsArray[index]['restaurantId'];
	}
	const name = state.restaurantsArray[index]['name'];
	try {
		const alreadyFavoritedResponse = await axios.post(`${baseUrl}/favorites`, {
			restaurantId: restaurantId,
			name: name
		});
		const alreadyRatedResponse = await axios.get(`${baseUrl}/${restaurantId}/review`);
		const categories = state.restaurantsArray[index]['categories'] ? state.restaurantsArray[index]['categories'].map((obj) => obj.title) : [];
		const categoriesString = categories.length ? categories.join(', ') : '';
		const newState = {
			...state,
			alreadyFavorited: alreadyFavoritedResponse.data[0]['contains'],
			alreadyRated: alreadyRatedResponse.data[0]['contains'],
			allCategories: categoriesString,
			restaurantsArray: [state.restaurantsArray[index]]
		};
		return newState;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const viewRestaurantsSlice = createSlice({
	name: 'viewRestaurantsAsync',
	initialState,
	reducers: {
		resetRestaurantRequestStatus: (state) => {
			state.currentState.restaurantRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(viewRestaurantsAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					restaurantInfo: payload,
					restaurantRequestStatus: 'fulfilled'
				});
			})
			.addCase(viewRestaurantsAsync.rejected, (state) => {
				state.currentState.restaurantRequestStatus = 'rejected';
			});
	}
});

export const { resetRestaurantRequestStatus } = viewRestaurantsSlice.actions;
export const selectRestaurantState = (state) => state.restaurant.currentState;

export default viewRestaurantsSlice.reducer;
