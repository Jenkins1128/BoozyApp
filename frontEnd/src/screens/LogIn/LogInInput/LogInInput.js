import React from 'react';
import { View, TextInput, Text, TouchableOpacity, StyleSheet } from 'react-native';

const LogInInput = ({ state, dispatch, updateEmail, updatePassword, goToSignup, loginPressed }) => {
	return (
		<View style={styles.inputContainer}>
			<TextInput style={styles.emailInput} onChangeText={(email) => dispatch(updateEmail({ email: email }))} value={state.email} placeholder='Email' placeholderTextColor='#FFFFFF' underlineColorAndroid='transparent'></TextInput>
			<TextInput
				style={styles.passwordInput}
				secureTextEntry={true}
				onChangeText={(password) => dispatch(updatePassword({ password: password }))}
				value={state.password}
				placeholder='Password'
				placeholderTextColor='#FFFFFF'
				underlineColorAndroid='transparent'
			></TextInput>
			<TouchableOpacity onPress={goToSignup} style={styles.signUpButton}>
				<Text style={styles.signUpText}>Sign Up</Text>
			</TouchableOpacity>
			<TouchableOpacity testID={'loginButton'} onPress={loginPressed} style={styles.loginButton}>
				<Text style={styles.logInText}>Log In</Text>
			</TouchableOpacity>
		</View>
	);
};

const styles = StyleSheet.create({
	inputContainer: {
		flex: 4
	},
	emailInput: {
		width: 320,
		borderRadius: 30,
		alignSelf: 'center',
		color: '#FFFFFF',
		marginBottom: 10,
		padding: 30,
		borderWidth: 1,
		borderColor: '#ededed'
	},
	passwordInput: {
		width: 320,
		borderRadius: 30,
		alignSelf: 'center',
		color: '#FFFFFF',
		padding: 30,
		marginBottom: 10,
		borderWidth: 1,
		borderColor: '#ededed'
	},
	signUpButton: {
		width: 150,
		height: 55,
		borderRadius: 70,
		alignItems: 'center',
		justifyContent: 'center',
		alignSelf: 'center',
		elevation: 8,
		color: '#fff',
		padding: 20,
		marginBottom: 10,
		backgroundColor: '#FFFFFF'
	},
	loginButton: {
		width: 240,
		height: 60,
		borderRadius: 70,
		alignItems: 'center',
		justifyContent: 'center',
		alignSelf: 'center',
		elevation: 8,
		color: '#fff',
		padding: 20,
		backgroundColor: '#FFFFFF'
	},
	logInText: {
		color: '#EB8873',
		fontSize: 14,
		fontWeight: 'bold'
	},
	signUpText: {
		color: '#EB8873',
		fontSize: 11,
		fontWeight: 'bold'
	}
});

export default LogInInput;
