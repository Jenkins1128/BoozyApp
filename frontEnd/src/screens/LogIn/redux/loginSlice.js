import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const loginAsync = createAsyncThunk('loginAsync/status', async (data, { rejectWithValue }) => {
	const { email, password } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/login', {
			email,
			password
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const loginSlice = createSlice({
	name: 'loginAsync',
	initialState,
	reducers: {
		updateEmail: (state, { payload }) => {
			state.currentState.email = payload.email;
		},
		updatePassword: (state, { payload }) => {
			state.currentState.password = payload.password;
		},
		resetStatus: (state) => {
			state.currentState.requestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(loginAsync.pending, (state) => {
				Object.assign(state.currentState, {
					requestStatus: 'pending',
					isLoading: true
				});
			})
			.addCase(loginAsync.fulfilled, (state) => {
				Object.assign(state.currentState, {
					requestStatus: 'fulfilled',
					isLoading: false
				});
			})
			.addCase(loginAsync.rejected, (state) => {
				Object.assign(state.currentState, {
					requestStatus: 'rejected',
					isLoading: false
				});
			});
	}
});

export const { updateEmail, updatePassword, resetStatus } = loginSlice.actions;
export const selectLogInState = (state) => state.login.currentState;

export default loginSlice.reducer;
