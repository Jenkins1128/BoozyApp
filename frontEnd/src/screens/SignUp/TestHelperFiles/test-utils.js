import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const INITIAL_STATE = {
	signup: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	},
	login: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	}
}; // Or use value from root reducer
const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
	render(ui, {
		wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
		...options
	});

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender, store };
