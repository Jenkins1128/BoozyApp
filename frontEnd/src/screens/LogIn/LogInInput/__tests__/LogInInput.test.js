import React from 'react';
import LogInInput from '../LogInInput';
import { render, cleanup, fireEvent } from '../../../../TestHelperFiles/redux/test-utils';

describe('<LogInInput />', () => {
	let state;
	let updateEmail, updatePassword, dispatch, loginPressed;
	let rendered;

	beforeEach(() => {
		state = {
			email: 'xxx@gmail.com',
			password: 'xxxxxx'
		};
		(updateEmail = jest.fn()), (updatePassword = jest.fn()), (dispatch = jest.fn()), (loginPressed = jest.fn());
		rendered = render(<LogInInput state={state} updateEmail={updateEmail} updatePassword={updatePassword} dispatch={dispatch} loginPressed={loginPressed} />);
	});
	afterEach(cleanup);

	//snapshot
	it('should render correctly', () => {
		expect(rendered).toMatchSnapshot();
	});

	//content
	it('should wrap inputs with a flexible wrapper', () => {
		const inputContainerComponent = rendered.getByTestId('inputContainer');
		const given = inputContainerComponent.props.style;
		const result = { flex: 4 };
		expect(result).toMatchObject(given);
	});

	it('should render emailInput styles and center itself', () => {
		const inputContainerComponent = rendered.getByTestId('emailInput');
		const given = inputContainerComponent.props.style;
		const result = {
			width: 320,
			borderRadius: 30,
			alignSelf: 'center',
			color: '#FFFFFF',
			marginBottom: 10,
			padding: 30,
			borderWidth: 1,
			borderColor: '#ededed'
		};
		expect(result).toMatchObject(given);
	});

	it('should render passwordInput styles and center itself', () => {
		const inputContainerComponent = rendered.getByTestId('passwordInput');
		const given = inputContainerComponent.props.style;
		const result = {
			width: 320,
			borderRadius: 30,
			alignSelf: 'center',
			color: '#FFFFFF',
			padding: 30,
			marginBottom: 10,
			borderWidth: 1,
			borderColor: '#ededed'
		};
		expect(result).toMatchObject(given);
	});

	it('should render goToSignup styles correctly', () => {
		const signUpButtonComponent = rendered.getByTestId('goToSignup');
		const given = signUpButtonComponent.props.style;
		const result = {
			width: 150,
			height: 55,
			borderRadius: 70,
			alignItems: 'center',
			justifyContent: 'center',
			alignSelf: 'center',
			elevation: 8,
			color: '#fff',
			padding: 20,
			marginBottom: 10,
			backgroundColor: '#FFFFFF',
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render signUpText styles correctly', () => {
		const signUpTextComponent = rendered.getByTestId('signUpText');
		const given = signUpTextComponent.props.style;
		const result = {
			color: '#EB8873',
			fontSize: 11,
			fontWeight: 'bold'
		};
		expect(result).toMatchObject(given);
	});

	it('should render loginButton styles correctly', () => {
		const loginButtonComponent = rendered.getByTestId('loginButton');
		const given = loginButtonComponent.props.style;
		const result = {
			width: 240,
			height: 60,
			borderRadius: 70,
			alignItems: 'center',
			justifyContent: 'center',
			alignSelf: 'center',
			elevation: 8,
			color: '#fff',
			padding: 20,
			backgroundColor: '#FFFFFF',
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render logInText styles correctly', () => {
		const logInTextComponent = rendered.getByTestId('logInText');
		const given = logInTextComponent.props.style;
		const result = {
			color: '#EB8873',
			fontSize: 14,
			fontWeight: 'bold'
		};
		expect(result).toMatchObject(given);
	});

	//interaction
	it('should fire updateEmail onChangeText event', () => {
		const inputComponent = rendered.getByTestId('emailInput');
		fireEvent(inputComponent, 'changeText', 'new text');
		expect(updateEmail).toHaveBeenCalledWith({ email: 'new text' });
	});

	it('should fire updatePassword onChangeText event', () => {
		const inputComponent = rendered.getByTestId('passwordInput');
		fireEvent(inputComponent, 'changeText', 'new text');
		expect(updatePassword).toHaveBeenCalledWith({ password: 'new text' });
	});

	it('should fire loginPressed event', () => {
		const loginPressedButton = rendered.getByText('Log In');
		fireEvent.press(loginPressedButton);
		expect(loginPressed).toHaveBeenCalledTimes(1);
	});
});
