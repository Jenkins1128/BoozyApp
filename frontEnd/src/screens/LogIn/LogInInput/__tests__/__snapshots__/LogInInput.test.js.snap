// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogInInput /> should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 4,
    }
  }
  testID="inputContainer"
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
    placeholderTextColor="#FFFFFF"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "#ededed",
        "borderRadius": 30,
        "borderWidth": 1,
        "color": "#FFFFFF",
        "marginBottom": 10,
        "padding": 30,
        "width": 320,
      }
    }
    testID="emailInput"
    underlineColorAndroid="transparent"
    value="xxx@gmail.com"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    placeholderTextColor="#FFFFFF"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "#ededed",
        "borderRadius": 30,
        "borderWidth": 1,
        "color": "#FFFFFF",
        "marginBottom": 10,
        "padding": 30,
        "width": 320,
      }
    }
    testID="passwordInput"
    underlineColorAndroid="transparent"
    value="xxxxxx"
  />
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 70,
        "color": "#fff",
        "elevation": 8,
        "height": 55,
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "padding": 20,
        "width": 150,
      }
    }
    testID="goToSignup"
  >
    <Text
      style={
        Object {
          "color": "#EB8873",
          "fontSize": 11,
          "fontWeight": "bold",
        }
      }
      testID="signUpText"
    >
      Sign Up
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 70,
        "color": "#fff",
        "elevation": 8,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 20,
        "width": 240,
      }
    }
    testID="loginButton"
  >
    <Text
      style={
        Object {
          "color": "#EB8873",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
      testID="logInText"
    >
      Log In
    </Text>
  </View>
</View>
`;
